cloudml-template/examples/german-custom/trainer/metadata.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ************************************************************************
+# YOU NEED TO MODIFY THE META DATA TO ADAPT THE TRAINER TEMPLATE YOUR DATA
+# ************************************************************************
+
+# task type can be either 'classification' or 'regression', based on the target feature in the dataset
+TASK_TYPE = 'custom'
+
+# list of all the columns (header) of the input data file(s)
+HEADER = ['A1', 'A2', 'A3', 'A4', 'A5', 'A6', 'A7', 'A8', 'A9', 'A10',
+          'A11', 'A12', 'A13', 'A14', 'A15', 'A16', 'A17', 'A18', 'A19', 'A20',
+          'class']
+
+# list of the default values of all the columns of the input data, to help decoding the data types of the columns
+HEADER_DEFAULTS = [['?'], [0.0], ['?'], ['?'], [0.0], ['?'], ['?'], [0.0], ['?'], ['?'],
```

Comment:
is the usage of `'?'` here conventional?  (assuming it means there is no default?)

Reply:
It doesn't matter which string value is used; can be 'NA', 'NULL', '?', or anything else...
It is just to till the tf csv decoder that this feature is a string.

Do you recommend specific value you use?

Reply:
Are the default values somehow used to fill in missing values, or only for specifying the type of the column?  I couldn't quite tell from the [docs](https://www.tensorflow.org/api_docs/python/tf/decode_csv).

I would guess that the type of `0.0` is `tf.float32` - is this correct?

Reply:
Yes you are right, you use `0.0` for `tf.float32`. You can check the census example (also in this cloudml-sample repo [here](https://github.com/GoogleCloudPlatform/cloudml-samples/blob/master/census/estimator/trainer/model.py#L29) line 29.

Reply:
If it is indeed the case that these "default" values serve only the purpose of communicating types, please consider using the empty string for the string type.  `'?'` appears to be some magical convention.